instrumentation/instrument.go

```diff
@@ -182,79 +179,23 @@ func InstrumentHandlerFuncWithOpts(opts prometheus.Opts, handlerFunc func(http.R
 		f.Inc()
 		defer f.Dec()
 
-		delegate := &responseWriterDelegator{ResponseWriter: w}
-		//out := computeApproximateRequestSize(r)
+		stats := stats{}
 
-		_, cn := w.(http.CloseNotifier)
-		_, fl := w.(http.Flusher)
-		_, hj := w.(http.Hijacker)
-		_, rf := w.(io.ReaderFrom)
-		var rw http.ResponseWriter
-		if cn && fl && hj && rf {
```

Comment:
That is exactly the goal. The problem with such runtime checks is that they are brittle and they silently downgrade the ResponseWriter preventing downstream handlers from using the complete ResponseWriter interface.